src/devices/Nmea0183/Sentences/EngineData.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnitsNet;
+
+namespace Iot.Device.Nmea0183.Sentences
+{
+    /// <summary>
+    /// A helper DTO to transfer engine data in one blob
+    /// </summary>
+    [Serializable]
+    public record EngineData
+    {
+        /// <summary>
+        /// Constructs an instance containing all relevant data
+        /// </summary>
+        public EngineData(int messageTimeStamp, int engineNo, RotationalSpeed revolutions, Ratio pitch, TimeSpan operatingTime, Temperature? engineTemperature)
+        {
+            MessageTimeStamp = messageTimeStamp;
+            EngineNo = engineNo;
+            Revolutions = revolutions;
+            Pitch = pitch;
+            OperatingTime = operatingTime;
+            EngineTemperature = engineTemperature;
+        }
+
+        /// <summary>
+        /// Constructs an <see cref="EngineData"/> instance from the two relevant messages
+        /// </summary>
+        /// <param name="fast">The fast-updating engine message</param>
+        /// <param name="detail">The slow updating engine message</param>
+        /// <returns>An <see cref="EngineData"/> instance</returns>
+        /// <exception cref="InvalidOperationException">The two messages are not from the same engine.</exception>
+        public static EngineData FromMessages(SeaSmartEngineFast fast, SeaSmartEngineDetail detail)
+        {
+            if (fast.EngineNumber != detail.EngineNumber)
+            {
+                throw new InvalidOperationException("The two messages are not for the same engine");
+            }
+
+            return new EngineData(fast.MessageTimeStamp, fast.EngineNumber, fast.RotationalSpeed, fast.PropellerPitch,
+                detail.OperatingTime, detail.Temperature);
+        }
+
+        /// <summary>
+        /// The NMEA2000 bus timestamp (not really relevant, I think)
+        /// </summary>
+        public int MessageTimeStamp
+        {
+            get;
+        }
```

Comment:
nit: I'd suggest more succint form in this file:

```suggestion
        public int MessageTimeStamp { get; }
```